pkg/gardenlet/controller/seed/seed.go

```diff
@@ -189,48 +188,25 @@ func (c *Controller) Run(ctx context.Context, workers int) {
 }
 
 func (c *Controller) startHealthManagement(ctx context.Context) {
```

Comment:
Please simplify the lease controller as well, i.e. remove the `leaseMap` in favor of a single `bool` value or some other simple mechanism.